app/src/main/kotlin/com/saltedge/authenticator/features/main/MainActivity.kt

```diff
@@ -21,263 +21,172 @@
-        } else {
-            snackbar = this.buildWarning(getString(R.string.warning_no_internet_connection))
-            snackbar?.show()
-        }
+    override fun getSnackbarAnchorView(): View? = container
+
+    private fun setupViewModel() {
+        viewModel = ViewModelProvider(this, viewModelFactory).get(MainActivityViewModel::class.java)
+        lifecycle.addObserver(viewModel)
+
+        // Events
+        viewModel.onQrScanClickEvent.observe(this, Observer {
+            it.getContentIfNotHandled()?.let {
+                this.showQrScannerActivity()
+            }
+        })
+        viewModel.onAppBarMenuClickEvent.observe(this, Observer {
+            it.getContentIfNotHandled()?.let {
+                //TODO show appbar menu
+            }
+        })
+        viewModel.onBackActionClickEvent.observe(this, Observer {
+            it.getContentIfNotHandled()?.let {
+                onBackPressed()
+            }
+        })
+        viewModel.onRestartActivityEvent.observe(this, Observer {
+            it.getContentIfNotHandled()?.let {
+                super.restartLockableActivity()
+            }
+        })
+        viewModel.onShowAuthorizationsListEvent.observe(this, Observer {
+            it.getContentIfNotHandled()?.let {
+                this.replaceFragmentInContainer(AuthorizationsListFragment())
+            }
+        })
+        viewModel.onShowAuthorizationDetailsEvent.observe(this, Observer {
+            it.getContentIfNotHandled()?.let { authorizationIdentifier ->
+                this.addFragment(AuthorizationDetailsFragment.newInstance(authorizationIdentifier))
+            }
+        })
+        viewModel.onShowConnectionsListEvent.observe(this, Observer {
+            it.getContentIfNotHandled()?.let {
+                this.addFragment(ConnectionsListFragment())
+            }
+        })
+        viewModel.onShowSettingsListEvent.observe(this, Observer {
+            it.getContentIfNotHandled()?.let {
+                this.addFragment(SettingsListFragment())
+            }
+        })
+        viewModel.onShowConnectEvent.observe(this, Observer {
+            it.getContentIfNotHandled()?.let { connectAppLinkData ->
+                this.addFragment(ConnectProviderFragment.newInstance(connectAppLinkData = connectAppLinkData))
+            }
+        })
+        viewModel.onShowSubmitActionEvent.observe(this, Observer {
+            it.getContentIfNotHandled()?.let {
+//        this.addFragment(
+//            SubmitActionFragment.newInstance(//TODO refactor new instance to exclude connection selection in main activity
+//                connectionGuid = connectionGuid,
+//                actionAppLinkData = actionAppLinkData
+//            )
+//        )
+                //TODO show submit action
+            }
+        })
+
+        // Views states
+        viewModel.internetConnectionWarningVisibility.observe(this, Observer {
+            //TODO update internet connection warning visibility
```

Comment:
U removed snackbar and added TODO, not the best solution I think

Reply:
I think it is the good solution, because we will change No Internet connection warning method from snackbar to empty screen state.

Reply:
but you didn't do it in the current pr